### ISSUE TITLE
Failure-tolerant + thoroughly tested Experiment Builder

### ISSUE THREAD

**Issue by Datamance** [lookit/lookit-api#574]
**TL;DR**: More robust Experiment Builder with a battery of tests to guard against regressions

**Narrative**
As a Developer, I want to have confidence that changes I push to settings and various services within our django app do not cause unexpected behavior/performance regressions in the experiment builder process (ex.: #571). I want my CI/CD pipeline to prove that I can be confident that the build process will remain functional when I deploy.

As a Researcher, I want to have more visibility into the build process itself (ex.: #571).

**Acceptance Criteria**

- [ ] New tests for ExperimentBuilder and EmberFrameplayerBuilder ensuring build errors are caught and handled gracefully
- [ ] Build phase broadcasting (probably a new field on the study model)
